### PR TITLE
STAF-36: fix layout, add NavBar component

### PR DESCRIPTION
**Pull request by RomanLavrunov**
Couldn't resolve path errors that appeared only when I commited changes.

NewTeamContainer => import { Team } from '../redux/TeamReducerState';
teamSlice => import { TeamReducerState } from './TeamReducerState';

It's probably somehow connected with the name change of Staff to Team, otherwise not sure why it has popped up just recently

### CODE REVIEW

File: src/core/components/NavMenu/NavMenu.tsx [hunk 1]
```diff
@@ -1,86 +1,48 @@
...
+import { IconButtonWithLabel } from '../buttons/IconButton';
+import { COLOR_VARIANTS } from '../../types/ColorVariants';
 import './NavMenu.css';
 
-export const NavMenu = (): JSX.Element => {
-  const { pathname } = useLocation();
-  const dispatch = useAppDispatch();
+interface NavMenuProps {
+  activeRoute: string;
+  icons: unknown;
```
Comment (devmc-ee) [inline src/core/components/NavMenu/NavMenu.tsx:11]: lets define icons as:
```
icons: { [name: string]: () => JSX.Element }
```

File: src/core/components/NavMenu/NavMenu.tsx [hunk 2]
```diff
@@ -1,86 +1,48 @@
...
+  const { t } = useTranslation();
 
-    if (route === pathname) return activeRouteClassName;
-
-    if (route === '/') return '';
-
-    const isActiveRoute = new RegExp(route, 'ui').test(pathname);
-
-    return isActiveRoute ? activeRouteClassName : '';
+  const getIcon = (iconName: unknown): JSX.Element => {
```
Comment (devmc-ee) [inline src/core/components/NavMenu/NavMenu.tsx:18]: iconName should be of string type

File: src/core/components/NavMenu/NavMenu.tsx [hunk 3]
```diff
@@ -1,86 +1,48 @@
...
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    //@ts-ignore icons type
+    return !!icons[iconName as keyof typeof icons] ? icons[iconName as keyof typeof icons]() : <HomeIcon />;
```
Comment (devmc-ee) [inline src/core/components/NavMenu/NavMenu.tsx:21]: If we define icons as` { [name: string]: () => JSX.Element }`  in the interface then the return statement can be simplified to 
`return !!icons[iconName] ? icons[iconName]() : <HomeIcon />;`

**Review by devmc-ee — CHANGES_REQUESTED** [review]

File: src/core/components/NavMenu/NavMenu.tsx (hunk 3 above)
Comment (RomanLavrunov) [reply src/core/components/NavMenu/NavMenu.tsx]: Changed.

File: src/core/components/NavMenu/NavMenu.tsx (hunk 2 above)
Comment (RomanLavrunov) [reply src/core/components/NavMenu/NavMenu.tsx]: Changed type to string.

File: src/core/components/NavMenu/NavMenu.tsx (hunk 1 above)
Comment (RomanLavrunov) [reply src/core/components/NavMenu/NavMenu.tsx]: Redefined icons' type.

File: src/config/routes/hooks.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+import { changeActiveRoute } from '../../core/redux/coreSlice';
+import { useAppDispatch } from '../redux/hooks';
+
+export const useUrlObserver = () => {
```
[the remark this answers is not in the document]
Comment (RomanLavrunov) [reply src/config/routes/hooks.ts]: Moved to core/hooks

File: src/App.tsx [hunk 5]
```diff
@@ -2,18 +2,22 @@ import './styles.css';
 import { Link, Route, Routes } from 'react-router-dom';
 import { Dashboard, Login } from './pages';
 import { ProtectedRoute } from './core/components/ProtectedRoute';
-import { routes } from './core/routing/routes';
-import { NewStaffPage, StaffPage } from './staff/components';
+import { PublicRoutesEnum, ProtectedRoutesEnum } from './config/routes/routes';
+import { NewTeamPage, TeamPage } from './team/components';
 import { RootNavLayout } from './core/components/RootNavLayout';
 import { ShiftsPage, NewShiftPage } from './shift/components';
+import { useUrlObserver } from './config/redux/hooks';
```
[the remark this answers is not in the document]
Comment (RomanLavrunov) [reply src/App.tsx]: Fixed

**Review by devmc-ee — APPROVED** [review]

**Pull request merged**